.pre-commit-config.yaml

```diff
@@ -1,6 +1,7 @@
 # https://pre-commit.com/
 ci:
     autoupdate_schedule: monthly
+exclude: 'xarray/datatree_.*'
```

Comment:
What fails here?

Ideally we want to promote exclusions into the tools themselves — like we have for mypy — rather than in the pre-commit-config. Since editors will run many of these by default, and will only pick up on exclusions from the tools' configs...

Though if someone wants to own fixing the issues soon after merging, no need to slow down the initial merge...

Reply:
`ruff` is the one that fails. However, `pre-commit` passes all (changed) files explicitly, so I'm not sure how easy it would be to exclude directly in `ruff` (we could add an additional `exclude` to `ruff`'s config, though).

Reply:
I was unable to generate an exclude for ruff's config only that actually seemed to be respected.  

In this exclude block https://github.com/flamingbear/xarray/blob/datatree-import/pyproject.toml#L233
I tried variations on "xarray/datatree_.*", "xarray/datatree_", and "datatree_"

I'm open to suggestions.